docs/understand/hardware_implementation.rst

```diff
@@ -0,0 +1,99 @@
+.. meta::
+  :description: This chapter describes the typical hardware implementation of GPUs supported by HIP.
+  :keywords: AMD, ROCm, HIP, Hardware, Compute Unit, ALU, Cache, Registers, LDS
+
+*******************************************************************************
+Hardware Implementation
+*******************************************************************************
+
+This chapter describes the typical hardware implementation of GPUs supported by
+HIP, and how the :ref:`inherent_thread_model` maps to the hardware.
+
+Compute Units
+===============================================================================
+
+The basic building block of a GPU is a so called Compute Unit (CU), also known
+as Streaming Multiprocessor (SM) on NVIDIA GPUs. The blocks making up a grid
+are scheduled for execution on the CUs. Each block is assigned to an individual
+CU, and a CU can accommodate several blocks. Depending on their resource usage
+up to thousands of threads can reside on a CU.
+
+Compute units contain an array of processing elements, also referred to as
+vector ALU (VALU), that execute the actual instructions of the threads,
+together with the necessary registers and caches.
+
+The threads of a block are executed in groupings called warps. The amount of
+threads making up a warp is architecture dependent.
+
+On AMD GPUs the warp size commonly is 64 threads, with the exception of RDNA
+based architectures, which can run in wave32 or wave64 mode, utilizing a warp
+size of 32 or 64 respectively. The warp size of each supported AMD GPU is
+listed in :doc:`rocm:reference/gpu-arch-specs`. NVIDIA GPUs have a warp size of
+32.
+
+In contrast to CPUs, GPUs generally don't employ complex cache structures or
+control logic, like branch prediction or out-of-order execution, but instead
+rely on massive hardware multithreading to hide latency.
+
+Context switching between warps residing on a compute unit incurs no overhead,
+as the context for the warps is stored on the compute unit and doesn't need to
+be fetched from memory. If there aren't enough free registers to accommodate
+all warps of a block, the block can't be scheduled to that compute unit and it
+has to wait until other blocks finish execution.
+
+The amount of warps that can reside on a compute unit concurrently, also known
+as occupancy, is determined by the warps resource usage, like registers and
```

Comment:
determined by the warps' resource usage

Reply:
You're right about the missing apostrophe. Changed to `warp's`, as this is meant to be singular.